src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs

```diff
@@ -95,13 +95,12 @@ public interface IChainedHeaderTree
         List<int> ConsensusTipChanged(ChainedHeader newConsensusTip);
 
         /// <summary>
-        /// Finds the header and verifies block integrity.
+        /// Finds the header for a given block.
         /// </summary>
         /// <param name="block">The block.</param>
         /// <returns>Chained header for a given block.</returns>
         /// <exception cref="BlockDownloadedForMissingChainedHeaderException">Thrown when block data is presented for a chained block that doesn't exist.</exception>
-        /// <exception cref="IntegrityValidationFailedException">Thrown in case integrity validation failed.</exception>
-        ChainedHeader FindHeaderAndVerifyBlockIntegrity(Block block);
+        ChainedHeader FindHeader(Block block);
```

Comment:
Maybe change from block to hash? 
CHT needs only hash and it looks like all the callers already have hash in the variable somewhere

Reply:
That wasn't really in the original scope, but I will do it if you want me to.

Reply:
Yes, please. 

I know that's extra but that would make perfect sense.